packages/argo-run/src/dev.ts

```diff
@@ -89,18 +91,115 @@ export async function dev(...args: string[]) {
+
         res.set('Content-Type', 'text/html');
-        res.send('<html>TODO</html>');
+        res.set('Cache-Control', 'no-cache');
+
+        function renderIndexPage(content: string) {
+          res.send(`
+            <html>
+              <head>
+                <meta data-react-html="true" name="viewport" content="width=device-width, initial-scale=1.0, height=device-height, viewport-fit=cover">
+                <style>
+                  * {
+                    box-sizing: border-box;
+                  }
+
+                  body {
+                    min-height: 100vh;
+                    min-height: -webkit-fill-available;
+                    display: grid;
+                    margin: 0;
+                    align-content: center;
+                    justify-content: center;
+
+                    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
+                      'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
+                      sans-serif;
+                  }
+
+                  html {
+                    height: -webkit-fill-available;
+                    background: black;
+                    color: white;
+                  }
+
+                  .content {
+                    max-width: 40rem;
+                    padding: 2rem;
+                    font-size: 1.5em;
+                  }
+
+                  code {
+                    color: mediumseagreen;
+                    word-break: keep-all;
+                  }
+                </style>
+              </head>
+
+              <body>
+                <div class="content">
+                  <p>
+                    This page is served by your local Argo development server. Instead of
+                    visiting this page directly, you will need to connect your local development
+                    environment to a real checkout environment.
+                  </p>
+                  <p>${content}</p>
+                </div>
+              </body>
+            </html>
+          `);
+        }
+
+        if (protocol === 'https') {
+          renderIndexPage(
+            `Create a checkout and append <code>?dev=${origin}/query</code> to the URL to start developing your extension.`,
+          );
+        } else {
+          renderIndexPage(`Make sure you have an secure URL for your local development server by running <code>shopify tunnel start --port=${port}</code>,
+            create a checkout, and append <code>?dev=https://TUNNEL_URL/query</code> to the URL, where <code>TUNNEL_URL</code> is
+            replaced with your own ngrok URL.`);
+        }
+
         res.end();
       });
 
+      // This endpoint is used by C1 to request information about the local workspace,
+      // including what extensions are available locally, and how to connect to the
+      // build server websocket for those extensions.
+      app.get('/query', (req, res) => {
```

Comment:
Should we make  this `/data` to be consistent with Admin?

Reply:
We already used `/data` for the post-purchase extension, and it has a different shape than what I'd like to provide here. We can probably move back to that route once we clean up the post-purchase dev flow, though personally I think `/data` is a rather unusual route name and `/query` leaves more room for making it a more powerful communication channel between the host app and CLI